### PR TITLE
Reworks the station charter so it requires an admin online.

### LINKED ISSUE #259: Charter needs an admin approve system
Our BYOND hub name was changed to a rule-breaking one for an extended duration of time due to no admins being online when this happened.
It's currently removed but it would be ideal if this asks online admins for confirmation, as the approve link system does.
If none are online, it should default to rejecting it, unlike the /tg/ one.

### PR DESCRIPTION
**Pull request by Super3222**
fixes #259
Pr: Charter now requires an admin. Even if the person changing the station's name is an admin.

:cl: Super3222
rscadd: Station Charters ARE BACK!
rscadd: The new charter is now admin dependent, meaning that the names you make up as captain will be sent to them.
rscadd: Without any admins, the charter will not work..
rscadd: Remember you only have one shot with your station name, so choose wisely.
rscadd: If you feel like an admin is being unfair and not letting you choose a PERFECTLY reasonable name, then clear it up with a complaint on the forums or something of the sort (like contacting the council).
/:cl:

### CODE REVIEW

**Comment by TheSadFox** [comment]
There goes the dream of creating Yiffstation 13.

**Comment by FluffySurvivor** [comment]
admin approval sucks, admins will deny every fun name even slightly out of place

**Comment by Super3222** [comment]
And then that admin will get logged for denying in almost every way.

Then you can take it to the forums, and get them in some trouble for being a wet blanket. When a name is denied it gets sent to two different types of logs (admin and game) and then messages the admins.

Changing the station name also changes our public hub name... so some jokes may not be allowed.

**Comment by george99g** [comment]
> admin approval sucks, admins will deny every fun name even slightly out of place

Our hub name was literally `pm for traitor` for an entire round.
We _need_ this.

File: code/game/objects/items/charter.dm [hunk 1]
```diff
@@ -2,24 +2,69 @@
...
 	icon_state = "scroll2"
-	desc = "An official document entrusting the governance of the station and surrounding space to the Captain. "
+	desc = "An official document entrusting the governance of the station and surrounding space to the Captain. Despite the fact it looks like a scroll, it actually has highly powerful electronic bindings attached which transmits information back and forth to Centcomm."
 	var/used = FALSE
+	var/pending_name
+	var/admin_controlled
+	var/mob/living/scripter
 
 /obj/item/station_charter/attack_self(mob/living/user)
 	..()
```
Comment (george99g) [inline code/game/objects/items/charter.dm:12]: Uhm. Might want to remove this.

Comment (Super3222) [inline code/game/objects/items/charter.dm:12]: I'm not sure what you're looking at.

Comment (oisin100) [inline code/game/objects/items/charter.dm:12]: ¯_(ツ)_/¯

File: code/game/objects/items/charter.dm [hunk 2]
```diff
@@ -2,24 +2,69 @@
...
 
-	//var/new_name = input(user, "What do you want to name [station_name()]? Keep in mind particularly terrible names may attract the attention of your employers.")  as text|null
-	//if(new_name)
-	//	world.name = new_name
-	//	station_name = new_name
-	//	minor_announce("[user.real_name] has designated your station as [world.name]", "Captain's Charter", 0)
+	var/new_name = input(user, "What do you want to name [station_name()]? Keep in mind particularly terrible names may attract the attention of your employers.")  as text|null
+	if(new_name)
+		pending_name = new_name
+		user << "You hear something crackle in your ears for a moment before a voice speaks.  \"Please stand by for a message on whether Central Command has approved your station's new name. Remember, you only get one chance. Message ends.\""
```
Comment (oisin100) [inline code/game/objects/items/charter.dm]: We really need a "You are now on hold" meme here with elevator music.

File: code/game/objects/items/charter.dm (hunk 1 above)
Comment (Super3222) [inline code/game/objects/items/charter.dm:12]: Oh, I see what you mean.

**Comment by FluffySurvivor** [comment]
>Our hub name was literally pm for traitor for an entire round.
We need this.

That's amazing

**Comment by FluffySurvivor** [comment]
> And then that admin will get logged for denying in almost every way.
> 
> Then you can take it to the forums, and get them in some trouble for being a wet blanket. When a name is denied it gets sent to two different types of logs (admin and game) and then messages the admins.

"hurr durr admin discretion bro" 

> Our hub name was literally pm for traitor for an entire round.

That's amazing

**Pull request merged**